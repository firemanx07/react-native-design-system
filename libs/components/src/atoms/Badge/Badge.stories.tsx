import { theme } from '@proxym/themes';
import { ComponentStory, Story } from '@storybook/react';
import React from 'react';

import {
  StorybookRow,
  StorybookScreen,
  StorybookSection,
} from '../../storybook';
import { DiamondIcon } from '../Icons';
import {
  Badge as BadgeBase,
  PropsType as BadgeProps,
  BadgeSize,
  BadgeVariant,
} from './Badge';

export default {
  component: BadgeBase,
  title: 'Atoms/Badge',
};

const roundedBadgeProps = {
  size: BadgeSize.Regular,
  variant: BadgeVariant.Rounded,
  children: '9',
};

const smallSquareBadgeProps = {
  size: BadgeSize.Small,
  variant: BadgeVariant.Square,
  children: 'NEW',
  style: { backgroundColor: theme.ds.colors.success },
  textStyle: { letterSpacing: 1.5 },
};

const regularSquareBadgeProps = {
  size: BadgeSize.Regular,
  variant: BadgeVariant.Square,
  children: 'Often chosen',
  style: { backgroundColor: theme.ds.colors.success },
};

const squareBadgeWithIconProps = {
  size: BadgeSize.Medium,
  variant: BadgeVariant.Square,
  contentColor: theme.ds.colors.primary,
  renderIconLeft: (size: number, color: string) => (
    <DiamondIcon width={size} height={size} fill={color} />
  ),
  children: 'Premium',
  style: { backgroundColor: theme.ds.colors.light },
};

const AllBadgesTemplate: Story = () => (
  <StorybookScreen title="All badges:">
    <StorybookRow>
      <StorybookSection title="Rounded Badge">
        <BadgeBase {...roundedBadgeProps} />
      </StorybookSection>
      <StorybookSection title="Small Square Badge">
        <BadgeBase {...smallSquareBadgeProps} />
      </StorybookSection>
      <StorybookSection title="Regular Square Badge">
        <BadgeBase {...regularSquareBadgeProps} />
      </StorybookSection>
    </StorybookRow>
    <StorybookRow>
      <StorybookSection title="Square Badge With Icon">
        <BadgeBase {...squareBadgeWithIconProps} />
      </StorybookSection>
    </StorybookRow>
  </StorybookScreen>
);

export const AllBadges = AllBadgesTemplate.bind({});

const Template: ComponentStory<typeof BadgeBase> = (args: BadgeProps) => (
  <StorybookScreen title="Badge:">
    <BadgeBase {...args} />
  </StorybookScreen>
);

const parameters = {
  controls: {
    include: ['size', 'children'],
  },
};

export const RoundedBadge = Template.bind({});
RoundedBadge.parameters = parameters;
RoundedBadge.args = roundedBadgeProps;

export const SmallSquareBadge = Template.bind({});
SmallSquareBadge.parameters = parameters;
SmallSquareBadge.args = smallSquareBadgeProps;

export const RegularSquareBadge = Template.bind({});
RegularSquareBadge.parameters = parameters;
RegularSquareBadge.args = regularSquareBadgeProps;

export const SquareBadgeWithIcon = Template.bind({});
SquareBadgeWithIcon.parameters = parameters;
SquareBadgeWithIcon.args = squareBadgeWithIconProps;
