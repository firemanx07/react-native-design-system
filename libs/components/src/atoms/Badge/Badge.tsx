import { styled, useNamespacedTheme } from '@proxym/themes';
import React, { memo, ReactNode, useMemo } from 'react';
import { StyleProp, ViewStyle, TextProps, TextStyle } from 'react-native';

import { TestIDType } from '../../types';
import { BaseText, TextVariant, TextWeight } from '../BaseText';

export enum BadgeSize {
  Regular = 24,
  Medium = 20,
  Small = 18,
}

export enum BadgeVariant {
  Rounded = 'Rounded',
  Square = 'Square',
  ArrowDown = 'ArrowDown',
}

export type PropsType = {
  children: TextProps['children'];
  variant?: BadgeVariant;
  size?: BadgeSize;
  renderIconLeft?: (size: number, color: string) => ReactNode;
  style?: StyleProp<ViewStyle>;
  contentColor?: string;
  textStyle?: StyleProp<TextStyle>;
} & TestIDType;

export const Badge = ({
  size = BadgeSize.Regular,
  variant = BadgeVariant.Rounded,
  children,
  style,
  renderIconLeft,
  contentColor,
  textStyle,
  testID,
}: PropsType) => {
  const { colors, iconSize } = useNamespacedTheme();
  const badgeContentColor = contentColor || colors.light;

  const textProps = useMemo(() => {
    switch (size) {
      case BadgeSize.Small:
        return {
          variant: TextVariant.XS,
          weight: TextWeight.ExtraBold,
        };
      case BadgeSize.Medium:
        return {
          variant: TextVariant.S,
          weight: TextWeight.Bold,
        };
      case BadgeSize.Regular:
      default:
        return {
          variant: TextVariant.M,
          weight: TextWeight.Bold,
        };
    }
  }, [size]);

  const borderRadius = useMemo(() => {
    switch (variant) {
      case BadgeVariant.Rounded:
        return size / 2;
      case BadgeVariant.Square:
      default:
        return 7;
    }
  }, [variant, size]);

  return (
    <Container
      size={size}
      borderRadius={borderRadius}
      style={style}
      testID={testID}
    >
      {!!renderIconLeft && (
        <IconWrapper>
          {renderIconLeft(iconSize.small, badgeContentColor)}
        </IconWrapper>
      )}
      <BadgeText color={badgeContentColor} {...textProps} style={textStyle}>
        {children}
      </BadgeText>
      {variant === BadgeVariant.ArrowDown && <ArrowDown style={style} />}
    </Container>
  );
};

type ContainerType = {
  size: BadgeSize;
  borderRadius: number;
};

const Container = styled.View<ContainerType>`
  height: ${({ size }) => size}px;
  min-width: ${({ size }) => size}px;
  border-radius: ${({ borderRadius }) => borderRadius}px;
  background-color: ${({ theme }) => theme.ds.colors.primary};
  justify-content: center;
  align-items: center;
  flex-direction: row;
  align-self: flex-start;
  padding: ${({ theme }) =>
    `${theme.ds.spacing.tiny}px ${theme.ds.spacing.small}px`};
`;

type BadgeTextType = {
  color?: string;
};

export const BadgeText = styled(BaseText)<BadgeTextType>`
  color: ${({ color }) => color};
`;

const IconWrapper = styled.View`
  padding-right: ${({ theme }) => theme.ds.spacing.tiny}px;
`;
const ArrowDown = styled.View<BadgeTextType>`
  width: 10px;
  height: 10px;
  background-color: ${({ theme }) => theme.ds.colors.primary};
  position: absolute;
  bottom: -4px;
  transform: rotate(45deg);
  border-radius: 2px;
  z-index: -1;
`;
export default memo(Badge);
