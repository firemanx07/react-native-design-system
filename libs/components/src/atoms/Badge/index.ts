export { default as Badge, BadgeSize, BadgeVariant } from './Badge';
