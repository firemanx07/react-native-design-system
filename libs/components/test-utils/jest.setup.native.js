import { Animated } from 'react-native';

require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();

console.warn = () => {};

Animated.timing = () => ({
  start: () => jest.fn(),
});
