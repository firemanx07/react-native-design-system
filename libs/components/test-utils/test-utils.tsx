import { theme, ThemeProvider } from '@celebrate-app/themes';
import { Story } from '@storybook/react';
import {
  cleanup,
  render,
  RenderOptions,
  waitFor,
} from '@testing-library/react-native';
import React, { ReactNode } from 'react';

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions
) => {
  return render(ui, { wrapper: Providers, ...options });
};

beforeEach(cleanup);
afterEach(cleanup);

export function testStory(displayName: string, TestComponent: Story) {
  test(`${displayName} snapshot test`, async () => {
    const comp = customRender(<TestComponent {...TestComponent.args} />);
    jest.runOnlyPendingTimers();

    await waitFor(() => {
      expect(comp).toBeTruthy();
      expect(comp.toJSON()).toMatchSnapshot();
    });
  });
}

export function testComponentSnapshot(
  TestComponent: React.ReactElement<
    any,
    string | React.JSXElementConstructor<any>
  >
) {
  test('Snapshot test', () => {
    const comp = customRender(TestComponent);
    expect(comp).toBeTruthy();
    expect(comp.toJSON()).toMatchSnapshot();
  });
}

export function testAllStories(stories: any) {
  Object.keys(stories).forEach((story) => {
    if (story !== 'default') {
      testStory(story, stories[story]);
    }
  });
}
