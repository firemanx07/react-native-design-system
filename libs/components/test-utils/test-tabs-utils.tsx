import { waitFor } from '@testing-library/react-native';

import { customRender } from './test-utils';

type renderTabsProps = {
  elem: JSX.Element;
  testIDs: string[];
};
export const renderTabs = ({ elem, testIDs }: renderTabsProps) => {
  const comp = customRender(elem);
  const tabs: ReturnType<typeof comp.getByTestId>[] = [];
  testIDs.forEach(testID => tabs.push(comp.getByTestId(testID)));
  return { comp, tabs };
};
export const testTabs = async (args: ReturnType<typeof renderTabs>) => {
  await waitFor(() => {
    expect(args.comp).toBeTruthy();
    args.tabs.forEach(tab => expect(tab).toBeDefined());
  });
};
