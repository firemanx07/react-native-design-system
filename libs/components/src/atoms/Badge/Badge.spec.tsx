import { testAllStories } from '../../../test-utils';
import * as stories from './Badge.stories';

describe('Badge tests', () => {
  testAllStories(stories);
});
