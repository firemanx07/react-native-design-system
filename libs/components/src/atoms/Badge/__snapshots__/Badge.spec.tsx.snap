// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge tests AllBadges snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#F2F3F4",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <Text
    fontFamily="Muli-Bold"
    fontSize={20}
    lineHeight={26}
    style={
      [
        {
          "color": "#222222",
          "fontFamily": "Muli-Bold",
          "fontSize": 20,
          "lineHeight": 26,
        },
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    All badges:
  </Text>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderColor": "#222222",
            "borderRadius": 4,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "marginBottom": 24,
            "marginLeft": 24,
            "marginRight": 24,
            "marginTop": 24,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 370,
          },
        ]
      }
      width={370}
    >
      <Text
        fontFamily="Muli-Bold"
        fontSize={16}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "backgroundColor": "#F2F3F4",
              "left": 16,
              "paddingHorizontal": 8,
              "position": "absolute",
              "top": -12,
            },
          ]
        }
      >
        Rounded Badge
      </Text>
      <View
        borderRadius={12}
        size={24}
        style={
          [
            {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#FF8066",
              "borderRadius": 12,
              "flexDirection": "row",
              "height": 24,
              "justifyContent": "center",
              "minWidth": 24,
              "paddingBottom": 4,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 4,
            },
          ]
        }
      >
        <Text
          color="#FFFFFF"
          fontFamily="Muli-Bold"
          fontSize={14}
          lineHeight={20}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          9
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "borderColor": "#222222",
            "borderRadius": 4,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "marginBottom": 24,
            "marginLeft": 24,
            "marginRight": 24,
            "marginTop": 24,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 370,
          },
        ]
      }
      width={370}
    >
      <Text
        fontFamily="Muli-Bold"
        fontSize={16}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "backgroundColor": "#F2F3F4",
              "left": 16,
              "paddingHorizontal": 8,
              "position": "absolute",
              "top": -12,
            },
          ]
        }
      >
        Small Square Badge
      </Text>
      <View
        borderRadius={4}
        size={18}
        style={
          [
            {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#FF8066",
              "borderRadius": 4,
              "flexDirection": "row",
              "height": 18,
              "justifyContent": "center",
              "minWidth": 18,
              "paddingBottom": 4,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 4,
            },
            {
              "backgroundColor": "#2FBFA7",
            },
          ]
        }
      >
        <Text
          color="#FFFFFF"
          fontFamily="Muli-ExtraBold"
          fontSize={10}
          lineHeight={14}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-ExtraBold",
                "fontSize": 10,
                "lineHeight": 14,
              },
              {
                "color": "#FFFFFF",
              },
              {
                "letterSpacing": 1.5,
              },
            ]
          }
        >
          NEW
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "borderColor": "#222222",
            "borderRadius": 4,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "marginBottom": 24,
            "marginLeft": 24,
            "marginRight": 24,
            "marginTop": 24,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 370,
          },
        ]
      }
      width={370}
    >
      <Text
        fontFamily="Muli-Bold"
        fontSize={16}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "backgroundColor": "#F2F3F4",
              "left": 16,
              "paddingHorizontal": 8,
              "position": "absolute",
              "top": -12,
            },
          ]
        }
      >
        Regular Square Badge
      </Text>
      <View
        borderRadius={4}
        size={24}
        style={
          [
            {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#FF8066",
              "borderRadius": 4,
              "flexDirection": "row",
              "height": 24,
              "justifyContent": "center",
              "minWidth": 24,
              "paddingBottom": 4,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 4,
            },
            {
              "backgroundColor": "#2FBFA7",
            },
          ]
        }
      >
        <Text
          color="#FFFFFF"
          fontFamily="Muli-Bold"
          fontSize={14}
          lineHeight={20}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          Often chosen
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderColor": "#222222",
            "borderRadius": 4,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "marginBottom": 24,
            "marginLeft": 24,
            "marginRight": 24,
            "marginTop": 24,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 370,
          },
        ]
      }
      width={370}
    >
      <Text
        fontFamily="Muli-Bold"
        fontSize={16}
        lineHeight={20}
        style={
          [
            {
              "color": "#222222",
              "fontFamily": "Muli-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "backgroundColor": "#F2F3F4",
              "left": 16,
              "paddingHorizontal": 8,
              "position": "absolute",
              "top": -12,
            },
          ]
        }
      >
        Square Badge With Icon
      </Text>
      <View
        borderRadius={4}
        size={20}
        style={
          [
            {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#FF8066",
              "borderRadius": 4,
              "flexDirection": "row",
              "height": 20,
              "justifyContent": "center",
              "minWidth": 20,
              "paddingBottom": 4,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 4,
            },
            {
              "backgroundColor": "#FFFFFF",
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "paddingRight": 4,
              },
            ]
          }
        >
          <test-file-stub
            fill="#FF8066"
            height={16}
            width={16}
          />
        </View>
        <Text
          color="#FF8066"
          fontFamily="Muli-Bold"
          fontSize={12}
          lineHeight={18}
          style={
            [
              {
                "color": "#222222",
                "fontFamily": "Muli-Bold",
                "fontSize": 12,
                "lineHeight": 18,
              },
              {
                "color": "#FF8066",
              },
            ]
          }
        >
          Premium
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Badge tests RegularSquareBadge snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#F2F3F4",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <Text
    fontFamily="Muli-Bold"
    fontSize={20}
    lineHeight={26}
    style={
      [
        {
          "color": "#222222",
          "fontFamily": "Muli-Bold",
          "fontSize": 20,
          "lineHeight": 26,
        },
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    Badge:
  </Text>
  <View
    borderRadius={4}
    size={24}
    style={
      [
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#FF8066",
          "borderRadius": 4,
          "flexDirection": "row",
          "height": 24,
          "justifyContent": "center",
          "minWidth": 24,
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
        },
        {
          "backgroundColor": "#2FBFA7",
        },
      ]
    }
  >
    <Text
      color="#FFFFFF"
      fontFamily="Muli-Bold"
      fontSize={14}
      lineHeight={20}
      style={
        [
          {
            "color": "#222222",
            "fontFamily": "Muli-Bold",
            "fontSize": 14,
            "lineHeight": 20,
          },
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      Often chosen
    </Text>
  </View>
</View>
`;

exports[`Badge tests RoundedBadge snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#F2F3F4",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <Text
    fontFamily="Muli-Bold"
    fontSize={20}
    lineHeight={26}
    style={
      [
        {
          "color": "#222222",
          "fontFamily": "Muli-Bold",
          "fontSize": 20,
          "lineHeight": 26,
        },
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    Badge:
  </Text>
  <View
    borderRadius={12}
    size={24}
    style={
      [
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#FF8066",
          "borderRadius": 12,
          "flexDirection": "row",
          "height": 24,
          "justifyContent": "center",
          "minWidth": 24,
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
        },
      ]
    }
  >
    <Text
      color="#FFFFFF"
      fontFamily="Muli-Bold"
      fontSize={14}
      lineHeight={20}
      style={
        [
          {
            "color": "#222222",
            "fontFamily": "Muli-Bold",
            "fontSize": 14,
            "lineHeight": 20,
          },
          {
            "color": "#FFFFFF",
          },
        ]
      }
    >
      9
    </Text>
  </View>
</View>
`;

exports[`Badge tests SmallSquareBadge snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#F2F3F4",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <Text
    fontFamily="Muli-Bold"
    fontSize={20}
    lineHeight={26}
    style={
      [
        {
          "color": "#222222",
          "fontFamily": "Muli-Bold",
          "fontSize": 20,
          "lineHeight": 26,
        },
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    Badge:
  </Text>
  <View
    borderRadius={4}
    size={18}
    style={
      [
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#FF8066",
          "borderRadius": 4,
          "flexDirection": "row",
          "height": 18,
          "justifyContent": "center",
          "minWidth": 18,
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
        },
        {
          "backgroundColor": "#2FBFA7",
        },
      ]
    }
  >
    <Text
      color="#FFFFFF"
      fontFamily="Muli-ExtraBold"
      fontSize={10}
      lineHeight={14}
      style={
        [
          {
            "color": "#222222",
            "fontFamily": "Muli-ExtraBold",
            "fontSize": 10,
            "lineHeight": 14,
          },
          {
            "color": "#FFFFFF",
          },
          {
            "letterSpacing": 1.5,
          },
        ]
      }
    >
      NEW
    </Text>
  </View>
</View>
`;

exports[`Badge tests SquareBadgeWithIcon snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#F2F3F4",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <Text
    fontFamily="Muli-Bold"
    fontSize={20}
    lineHeight={26}
    style={
      [
        {
          "color": "#222222",
          "fontFamily": "Muli-Bold",
          "fontSize": 20,
          "lineHeight": 26,
        },
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    Badge:
  </Text>
  <View
    borderRadius={4}
    size={20}
    style={
      [
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "backgroundColor": "#FF8066",
          "borderRadius": 4,
          "flexDirection": "row",
          "height": 20,
          "justifyContent": "center",
          "minWidth": 20,
          "paddingBottom": 4,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 4,
        },
        {
          "backgroundColor": "#FFFFFF",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "paddingRight": 4,
          },
        ]
      }
    >
      <test-file-stub
        fill="#FF8066"
        height={16}
        width={16}
      />
    </View>
    <Text
      color="#FF8066"
      fontFamily="Muli-Bold"
      fontSize={12}
      lineHeight={18}
      style={
        [
          {
            "color": "#222222",
            "fontFamily": "Muli-Bold",
            "fontSize": 12,
            "lineHeight": 18,
          },
          {
            "color": "#FF8066",
          },
        ]
      }
    >
      Premium
    </Text>
  </View>
</View>
`;
